src/kirby/components/list/list-item-color.directive.ts

```diff
@@ -0,0 +1,22 @@
+import { Directive, Input, Renderer2, ElementRef, OnChanges, SimpleChanges } from '@angular/core';
+
+@Directive({
+  selector: '[kirbyListItemColor]',
+})
+export class ListItemColorDirective implements OnChanges {
```

Comment:
Could this be merged with the existing themeColor directive? E.g. `color: (item: any) => string | string;`

Reply:
No I don't think so. They do different things. This calls the getItemColor cb and sets class and the other just maps a color input to a class to append to the host element.

Reply:
The plus side by merging it into the `ThemeColor` directive would be, that all handling of colors would be in one place. And the cool thing with the `ThemeColor` directive is, that the component has to handle what colors it's supports in its own styling.

So in the case of the list, it would have in its `scss` file:

```scss
:host(.white) {
...
}

:host(.light) {
...
}
```

And as the `item` also is user defined, i see no problem in merging this into the `ThemeColor` directive. So please reconsider it 😄

Reply:
But the color cb is only used in the list and the themeColor directive is used in multiple different places. Don't we pollute the themeColor directive by adding list specific logic to it? I know the cb is user defined, but still, it is created for the list.